src/index.js

```diff
@@ -24,17 +36,180 @@ function openTab(i) {
-
-// loop to assign eventListeners to all the tabs
 for (let index = 0; index < tab.length; index++) {
     const element = tab[index];
-    element.addEventListener('click', () => {
+    showTab(element, index);
+}
+
+// function to make tabs clickable
+function showTab(e,i) {
+    e.addEventListener('click', () => {
         clearTabs();
-        openTab(index);
+        openTab(i);
     });
-}
\ No newline at end of file
+}
+
+// loop to assign eventListeners to all the tabs
+// for (let index = 0; index < tab.length; index++) {
+//     const element = tab[index];
+//     element.addEventListener('click', () => {
+//         clearTabs();
+//         openTab(index);
+//     });
+// }
+
+function generateText(){
+    // Generating typing content
+    const paragraph = _paragraph(50);
+    content.innerHTML = paragraph;
+    // console.log(paragraph);
+
+    // To convert all the charecters to lower case
+    return smash = content.innerText;
+}
+
+function countDown(){
+    console.log('countdown started');
+    let haha = setInterval(() => {
+        // this has a delay of 1s, to compensate it time is shown as 'time-1' and only timeValue-1 intervals are requested.
+        if(time>1){
+            timeCount.innerHTML = (time-1);
+            time--;
+        }
+        else{
+            speed = w*2;
+            accuracy = Math.floor((i/k)*100);
+
+            timeCount.style.fontSize = '4rem';
+            timeCount.innerHTML = 'Typing Speed: '+ speed + 'WPM' + '</br>Accuracy: ' + accuracy + '%';
+
+            typeUpdate.disabled= true;
+            startButton.style.opacity = '1';
+            startButton.style.pointerEvents = 'auto';
+
+            clearInterval(haha);
+
+            // typeUpdate.removeEventListener('mousemove', validation);
+
+            time= timeValue;
+            startButton.disabled = false;
+            console.log(typeUpdate);
+        }
+    }, 1000);
+}
+
+function resetVariables(){
+    i= 0, j= 0, k= 0, w = 0, speed = 0, accuracy = 0;
+}
+
+// Verification of the entries (Did not work when outside of the event listner).
+// let validation = (e, f) => {
+//     let a = f[i];
+//     let b = e.key;
+//     console.log(a, b, i, "errors = " + j, k, w);
+//     k++;
+//     if(a==b){
+//         if(b == ' '){
+//             w++;
+//         };
+//         i++;
+//     }
+//     else if(b == 'Backspace'){
+//         if(b == ' '){
+//             w--;
+//         }
+//         i--;
+//     }
+//     else if(b == 'Shift' || b == 'Tab' || b == 'Alt'){
+//         console.log('hello you pressed an odd key');
+//     }
+//     else if (a != b){
+//         i++;
+//         j++;
+//     }
+// };
+
+window.onload = () => {
+    restartButton.style.display = 'none';
+    typeUpdate.disabled= true;
+
+    // generateText();
+
+    // to clear all the tabs when the page is loaded
+    clearTabs();
+
+    // to open the default tab wihch is tab[0]
+    openTab(0);
+}
+
+startButton.addEventListener('click', () => {
```

Comment:
fix: Let's prefer a named function for event listeners. Major Reasons:
1. Code reusability
2. Easy way to remove the event listener
3. Readable code